compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7025,6 +7037,30 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
         }
     }
 
+    private void populateIncludedRecordParams(BVarSymbol param, HashSet<String> includedRecordFields,
+                                              List<String> includedRecordParamNames) {
+        Set<String> fields = ((BRecordType) Types.getReferredType(param.type)).fields.keySet();
+        fields.removeIf(field -> !includedRecordParamNames.contains(field));
```

Comment:
This will remove the field from the underlying `BRecordType` also since we are operating on the keys returned by `keySet()`.

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html#keySet()

Reply:
Changed.